### PR TITLE
refactor: homepage and current component fixes

### PR DESCRIPTION
**Pull request by maciejkrol18**
This PR aims to properly set up the homepage and fix issues with all of the currently added components, therefore this PR will also serve as a form of a code review. Here's the detailed list of my added changes and the reasoning behind them:

## Minor changes
- Formatted all files in `src` directory with Prettier, applying styling rules defined in `.prettierrc`
- `layout.tsx`
  - changed the default title and description meta values
  - changed html tag's language to `pl` - afaik leaving it at `en` despite the page's content being in polish could negatively impact SEO in the future
- Fixed missing `key` prop in my DetailsSection component

## Homepage (https://github.com/Solvro/web-promochator/commit/2d744478c3362cd5d3ccea6eb34365b3b2b1f637)
The hero section and details section components were added to the homepage. Removed `h-full` and `w-full` classes from the "root" container. We don't need to specify full width, as any container with `display: block`, which is the default, takes up all of the available horizontal space. `h-full` doesn't even work here, because it sets `height` to `100%` - that value needs to be relative to something, i.e. the parent component needs to have a explicit height set in order for percentage based values to work for the child.

## Hero section (https://github.com/Solvro/web-promochator/commit/2d744478c3362cd5d3ccea6eb34365b3b2b1f637, https://github.com/Solvro/web-promochator/commit/cbf88b258e3eb4d554983e8bee97d861a5db8840)
**Line 12**
Removed `h-full`, `w-full` and `lg:h-screen` classes. Full width and height for same reasons for above, as for `lg:h-screen`, I assume it was set so that the hero section takes up the entire screen size in the initial view.
1. This doesn't take the navbar height into account.
2. Isn't really needed at this point anyway since hero section's content is big enough to take up the entire initial viewport

**Line 45**
I think we should strive towards reducing the amount of arbitrary values used within tailwind classes, I replaced that one with  `text-textSecondary` since it's basically almost the same color. Ultimately, I think we should decide how the color values are going to be setup in our project. Since we're using `shadcn/ui`, the best solution in my opinion, would be to override all of the default values in `globals.css` so that they adhere to our Figma design, and refer to them even outside of shadcn components specific code in the future

**Note: Line 55**
I think it's a leftover from an attempt at adding animations. I decided to not remove it as they do not affect the layout, but I think such things should not be kept if not utilized. Same applies for all the `relative` classes in adjacent containers, which I assume were needed for the animation to work

**Lines 64, 70**
Changed `h3` tags to `p`. Previous setup was against accessibility standards - in HTML we have heading tags from `h1` to `h6`, all defining title headings of different sections of the page. Going forward, I think it's crucial to remember this when writing future components for the homepage. For example, in my DetailsSection component, I used a `h3` tag for the "Jak ci pomożemy?" line, since it's the 3rd informational section on the page as per the Figma design.

**Line 77**
Removed the alt prop from the image, as it serves no informational purpose and functions as a decoration (and was also set to the image's filename) (as per [solvro's webdev handbook](https://docs.solvro.pl/guides/web-development/accessibility/#kiedy-nie-stosowa%C4%87))

## Navbar (https://github.com/Solvro/web-promochator/commit/df3279a97f2df59b288c6bbea53b485d3de5aecc)

Previously, the logo in the navbar was made with an image of the cubes and a paragraph put next to it. I exported the logo from the Figma design as a `.svg` file. Now instead of a flexbox div with an image and a paragraph inside of it, we just have a single `<Image/>` component, as I feel it's a more practical solution

**Line 13**
Uneccessary `w-full` class, as explained above
Unnecessary `flex-row`, as it's the default flexbox flow setting
Unnecessary `gap-12`, since we already have `justify-between` set, effectively negating any gap

**Line 31**
Removed `ml-auto` class, not sure what it was needed for, I don't think removing it broke anything, but removing it was crucial for `justify-between` to properly work in the root container.

<hr/>

That would be all for my changes. Please tell if I need to elaborate further on any of my changes in case I didn't explain them very well, and if anyone has any issues with them.

### CODE REVIEW

**Comment by chewmanji** [comment]
> Line 45
I think we should strive towards reducing the amount of arbitrary values used within tailwind classes, I replaced that one with text-textSecondary since it's basically almost the same color. Ultimately, I think we should decide how the color values are going to be setup in our project. Since we're using shadcn/ui, the best solution in my opinion, would be to override all of the default values in globals.css so that they adhere to our Figma design, and refer to them even outside of shadcn components specific code in the future

Imo, we would add colors from Figma's color palette to tailwind config/globals.css, not change shadcn colors for now and just use these colors defined by us.

File: src/components/DetailsSection.tsx [hunk 1]
```diff
@@ -17,47 +17,56 @@ function FeatureCard({icon, title, description}: FeatureCardProps) {
...
       title: "Zapis rekomendacji",
-      description: "Po rozmowie z naszym systemem możesz zapisać istotne dla ciebie rekomendacje wewnątrz aplikacji."
-    }
+      description:
+        "Po rozmowie z naszym systemem możesz zapisać istotne dla ciebie rekomendacje wewnątrz aplikacji.",
+    },
   ];
 
   return (
     <div className="flex flex-col gap-6 items-center justify-center text-center">
```
Comment (chewmanji) [inline src/components/DetailsSection.tsx]: How about replacing div with section with id attribute?

File: src/components/HeroSection.tsx [hunk 2]
```diff
@@ -1,51 +1,84 @@
...
-                                    <h3 className="text-[1.375rem] font-semibold pt-2"> Dr Jan Kowalski</h3>
-                                    <p>Specjalistka w dziedzinie UX Design</p>
-                                </div>
-                            </div>
-                            <Image src={ChatBot} alt="ChatBot-image" className="max-w-xs" />
-                        </div>
-                    </div>
+  return (
+    <>
+      <div className="flex flex-col items-center justify-center py-24 px-8 z-10">
```
Comment (chewmanji) [inline src/components/HeroSection.tsx]: Is empty fragment necessary here?
How about replacing div with section with id attribute so to enable navigating?

**Comment by mejsiejdev** [comment]
I do not really have anything to add, everything seems fine to me.

File: src/components/Navbar.tsx [hunk 3]
```diff
@@ -5,30 +5,25 @@ import { useState } from "react";
...
-          />
-          <p className="text-lg font-semibold">
-            Promo<span className="text-[#5f7ecd]">CHAT</span>or
-          </p>
-        </div>
+      <div className="flex justify-between items-center">
+        <Image
+          src={PromochatorLogo}
+          alt="logo Promochatora"
+          width={128}
```
Comment (muclx) [inline src/components/Navbar.tsx]: PROMOCHATOR logo is too small. Ideally, the size would be equal to the KN SOLVRO logo.

Comment (maciejkrol18) [reply src/components/Navbar.tsx]: I agree. I made a new Figma export without the padding and changed the Image tag sizes from `128x64` to `160x34` in https://github.com/Solvro/web-promochator/pull/17/commits/8a8a4c2b6ce00645b71e26c4ab894545d6422d61. I think it looks better now

File: src/components/HeroSection.tsx (hunk 2 above)
Comment (maciejkrol18) [reply src/components/HeroSection.tsx]: Good catch. Removed the empty fragment and made the div a section tag with id of "hero" in https://github.com/Solvro/web-promochator/pull/17/commits/8a8a4c2b6ce00645b71e26c4ab894545d6422d61

File: src/components/DetailsSection.tsx (hunk 1 above)
Comment (maciejkrol18) [reply src/components/DetailsSection.tsx]: Changed in https://github.com/Solvro/web-promochator/pull/17/commits/8a8a4c2b6ce00645b71e26c4ab894545d6422d61

File: src/components/HeroSection.tsx [hunk 4]
```diff
@@ -1,51 +1,82 @@
...
-                                </div>
-                            </div>
-                            <Image src={ChatBot} alt="ChatBot-image" className="max-w-xs" />
-                        </div>
-                    </div>
+  return (
+      <section id="hero" className="flex flex-col items-center justify-center py-24 px-8 z-10">
+        <div className="flex flex-col max-w-screen-lg text-center gap-8 lg:gap-12  lg:mt-8">
+          <div>
+            <h1 className="text-[clamp(2.7rem,5cqw,5rem)] font-bold leading-tight">
```
Comment (Rei-x) [inline src/components/HeroSection.tsx:14]: damn, cqw i clamp? widzę, że ktoś tu odrobił lekcje z css'a 😭

jak na landing to jak najbardziej spoko ale staraj się nie hardcodować wielkości, tutaj np 3xl i 7xl też by siadło zamiast tych remów

File: src/components/HeroSection.tsx [hunk 5]
```diff
@@ -1,51 +1,82 @@
...
+              {" "}
+              Nasz system rekomendacyjny dopasuje go do tytułu lub opisu twojej
+              pracy dyplomowej.{" "}
+            </p>
+          </div>
+          <div className="flex flex-col md:flex-row justify-center gap-4 text-[1.125rem]">
+            <Button
+              asChild
+              variant={"secondary"}
+              className="text-[1.125rem] px-6 py-[1.84rem] rounded-xl"
```
Comment (Rei-x) [inline src/components/HeroSection.tsx]: czemu taki padding na pixele?
text-lg to to samo co text-[1.125rem] z tego co kojarze
```suggestion
              className="text-lg px-6 py-[1.84rem] rounded-xl"
```

File: src/components/HeroSection.tsx [hunk 6]
```diff
@@ -1,51 +1,82 @@
...
+              <p>
+                {" "}
+                Wpływ modernistycznego stylu w projektowaniu stron
+                internetowychna percepcję i doświadczenia użytkownika.
+              </p>
+            </div>
+            <div className="flex lg:flex-row lg:items-start items-center flex-col-reverse justify-center">
+              <div className="relative lg:mt-6">
+                <div className="relative container bg-backgroundChatBot max-w-xs h-auto rounded-3xl  px-5 py-4 mt-4">
+                  <div className="absolute left-1/2 top-[-1.25rem] lg:left-[96%] lg:top-1/3 w-0 h-0 border-l-[1.25rem] border-r-[1.25rem] border-b-[1.25rem] border-transparent border-b-colorChatBot lg:transform lg:rotate-90"></div>
```
Comment (Rei-x) [inline src/components/HeroSection.tsx]: nie jestem fanem hardocodowania wartości jak 1.25 rem 😭

File: src/components/HeroSection.tsx [hunk 7]
```diff
@@ -1,51 +1,82 @@
...
+                  <p>
+                    Oto{" "}
+                    <span className="text-colorPrimary font-semibold">
```
Comment (Rei-x) [inline src/components/HeroSection.tsx]: ale dziwna zmienna tailwindowa, raczej się używa kebab-case tutaj

File: src/components/Navbar.tsx [hunk 8]
```diff
@@ -1,37 +1,29 @@
...
-            Promo<span className="text-[#5f7ecd]">CHAT</span>or
-          </p>
-        </div>
+      <div className="flex justify-between items-center">
+        <Image
+          src={PromochatorLogo}
+          alt="logo Promochatora"
+          width={160}
+          height={34}
+        />
```
Comment (Rei-x) [inline src/components/Navbar.tsx:20]: bardziej jako protip, lubię logo eksportować sobie w png i wtedy nie trzeba width i height podawać, bo nextjs sam sobie to ogarnia

na stronach internetowych też czasami png mniej waży niż svg, a różnicy w jakości nikt nie widzi 😭

**Review by Rei-x — APPROVED** [review]
super kodzik approve

na pewno bym się zastanowił czy chcesz tyle wartości hardkodować w css'ie, wątpię, że pixel perfect to jest coś co was interesuje i nie bałbym się odchodzić lekko od designów, po to, żeby kod był ładniejszy i to by to zajęło mniej czasu

// edit. sory, zapomniałem że wy po angielsku 😭

File: src/components/Navbar.tsx (hunk 8 above)
Comment (maciejkrol18) [reply src/components/Navbar.tsx:20]: logo w .svg waży 6.44KB a w .png 1.87KB, osobiście uważam że jest widoczna różnica w jakości - .png jest zdecydowanie lekko zblurowane, myślę że nie jest to aż taka różnica w rozmiarze żeby podmieniać, a jakość jest lepsza

File: src/components/HeroSection.tsx (hunk 7 above)
Comment (maciejkrol18) [reply src/components/HeroSection.tsx]: Podmieniłem wszystkie camelCase na kebab-case w tailwindowych zmiennych w https://github.com/Solvro/web-promochator/pull/17/commits/86d9b4371299e831f55fadf99a3ff14841a3b925

File: src/components/Navbar.tsx (hunk 8 above)
Comment (Rei-x) [reply src/components/Navbar.tsx:20]: no spoko, po prostu, żebyś wiedział, że używanie png to też jest poprawna opcja, bo sporo osób się fiksuje na tym, że logo = svg i zapomina o tym, że można inaczej

File: src/components/HeroSection.tsx (hunk 7 above)
Comment (Rei-x) [reply src/components/HeroSection.tsx]: epicko

File: src/components/HeroSection.tsx (hunk 5 above)
Comment (maciejkrol18) [reply src/components/HeroSection.tsx]: Ogarnięte w https://github.com/Solvro/web-promochator/pull/17/commits/e3773143b6862a5094b27f4d884752d12e00e390

File: src/components/HeroSection.tsx (hunk 6 above)
Comment (maciejkrol18) [reply src/components/HeroSection.tsx]: Tak jak napisałem w opisie PR'a, ten div wydaje mi się pozostałością z tego jak była próba robienia jakichś animacji więc póki co to po prostu wykomentowałem w https://github.com/Solvro/web-promochator/pull/17/commits/e3773143b6862a5094b27f4d884752d12e00e390

File: src/components/HeroSection.tsx (hunk 4 above)
Comment (maciejkrol18) [reply src/components/HeroSection.tsx:14]: Jak może być na landing to zostawiam

**Pull request merged**